src/mixins/indexMapper.js

```diff
@@ -0,0 +1,190 @@
+import { arrayReduce, arrayMap, arrayMax } from './../helpers/array';
+import { defineGetter } from './../helpers/object';
+import { rangeEach } from './../helpers/number';
+
+const MIXIN_NAME = 'indexMapper';
+
+/**
+ * @type {Object}
+ */
+const indexMapper = {
+  /**
+   * Array is storing indexes in the below form:
+   *
+   * VISUAL_INDEX¹ => PHYSICAL_INDEX²
+   *
+   * It maps from visible row / column to its representation in the data source. For example, when we sorted data, our 1st visible row can represent 4th row from the original data,
+   * 2nd can be mapped to 3rd, 3rd to 2nd, etc. (keep in mind that array is indexed from the zero).
+   *
+   * ¹ internally stored as array index; from 0 to n, where n is number of visible cells on the axis
+   * ² internally stored as array value, contains some indexes (just for not trimmed cells) from data source
+   *
+   */
+  _arrayMap: [],
+
+  /**
+   * Get physical index by its visual index.
+   *
+   * @param {Number} visualIndex Visual index.
+   * @return {Number|null} Returns translated index mapped by passed visual index.
+   */
+  getPhysicalIndex(visualIndex) {
+    const length = this._arrayMap.length;
+    let physicalIndex = null;
+
+    if (visualIndex < length) {
+      physicalIndex = this._arrayMap[visualIndex];
+    }
+
+    return physicalIndex;
+  },
+
+  /**
+   * Get visual index by its physical index.
+   *
+   * @param {Number} physicalIndex Physical index to search.
+   * @returns {Number|null} Returns a visual index of the index mapper.
+   */
+  getVisualIndex(physicalIndex) {
+    let visualIndex = null;
+
+    if (this._arrayMap.includes(physicalIndex)) {
```

Comment:
Won't `includes` and `indexOf` iterate over the array twice? Depends how build optimization will handle this.
```suggestion
    let index = this._arrayMap.indexOf(physicalIndex);
    if (index != -1) {
      visualIndex = index;
    }
```